src/renderer/pages/Staking/Overview/components/NominatorsList/ShardedList/ShardedList.tsx

```diff
@@ -0,0 +1,201 @@
+import { NominatorsItem } from '../NominatorItem/NominatorItem';
+
+type NominatorInfoWithShard = NominatorInfo<ShardAccount>;
+
+type Props = {
+  shardsStake: NominatorInfoWithShard[];
+  era?: number;
+  asset?: Asset;
+  explorers?: Explorer[];
+  isStakingLoading: boolean;
+  onCheckValidators: (stash?: Address) => void;
+  onToggleNominator: (nominator: Address, value?: boolean) => void;
+  getContent: (stake: NominatorInfoWithShard) => ReactNode;
+};
+
+export const ShardedList = ({
+  shardsStake,
+  era,
+  asset,
+  explorers,
+  isStakingLoading,
+  onCheckValidators,
+  onToggleNominator,
+  getContent,
+}: Props) => {
+  const { getNextUnstakingEra, hasRedeem } = useStakingData();
+
+  const { t } = useI18n();
+  const [shards, setShards] = useState<NominatorInfoWithShard[]>(shardsStake);
+  const [selectedShards, setSelectedShards] = useState<number>(0);
+  const [withdrawnShards, setWithdrawnShards] = useState<number>(0);
+  const [unstakingShards, setUnstakingShards] = useState<number>(0);
+
+  useEffect(() => {
+    setShards(shardsStake);
+  }, [shardsStake]);
+
+  const selectAllShards = (value: boolean) => {
+    const allShards = shards.map((shard) => {
+      onToggleNominator(shard.address, value);
+      setSelectedShards(value ? shards.length : 0);
+
+      return { ...shard, isSelected: value };
+    });
+
+    setShards(allShards);
+  };
+
+  const selectShard = (nominator: string, isChecked: boolean) => {
+    setSelectedShards((prev) => (isChecked ? prev + 1 : prev - 1));
+    onToggleNominator(nominator);
+  };
+
+  const getShards = (shardsStake: NominatorInfoWithShard[]) => {
+    if (!shards.length) return;
+
+    return shards.map((shard: NominatorInfoWithShard) => {
+      if (hasRedeem(shard.unlocking, era)) {
+        setWithdrawnShards((prev) => prev + 1);
+      }
+      if (getNextUnstakingEra(shard.unlocking, era)) {
+        setUnstakingShards((prev) => prev + 1);
+      }
+
+      return (
+        <li key={shard.account.id}>
+          <NominatorsItem
+            key={shard.account.id}
+            isStakingLoading={isStakingLoading}
+            content={getContent(shard)}
+            stake={shard}
+            nominatorsLength={shardsStake.length}
+            asset={asset}
+            explorers={explorers}
+            onToggleNominator={selectShard}
+            onCheckValidators={onCheckValidators}
+          />
+        </li>
+      );
+    });
+  };
+
+  const { totalReward, totalStake } = useMemo(() => {
+    const total = shards.reduce(
+      (acc, shard) => {
+        return {
+          totalReward: acc.totalReward + +(shard.totalReward || 0),
+          totalStake: acc.totalStake + +(shard.totalStake || 0),
+        };
+      },
+      { totalReward: 0, totalStake: 0 },
+    );
+
+    return { totalReward: total.totalReward.toString(), totalStake: total.totalStake.toString() };
+  }, [shards.length, shards]);
+
+  return (
+    <li key={shardsStake[0].account.groupId}>
```

Comment:
I assume we can move `<li key={ ... }>` outside of `ShardedList` into `NominatorsList`